src/parsers/pulldown.rs

```diff
@@ -17,15 +19,15 @@ pub struct ScrollableCache {
 /// Parse events until a desired end tag is reached or no more events are found.
 /// This is needed for multiple events that must be rendered inside a single widget
 fn delayed_events<'e>(
-    events: &mut impl Iterator<Item = (usize, pulldown_cmark::Event<'e>)>,
+    events: &mut impl Iterator<Item = (usize, (pulldown_cmark::Event<'e>, Range<usize>))>,
```

Comment:
Maybe there could be a type alias for this, but I wasn't sure enough about it to make it one.

Reply:
I thought the exact same thing. I have some refactoring to do :)